### PR TITLE
[xray] 3.4.0 release

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
* Update Xray to version `3.4.0` - https://www.jfrog.com/confluence/display/JFROG/Xray+Release+Notes#XrayReleaseNotes-Xray3.4
* Added Upgrade Notes in README for 3.x upgrades - https://github.com/jfrog/charts/blob/master/stable/mission-control/README.md#special-upgrade-notes
* Bump router version to `1.4.0`
* Bump postgresql tag version to `9.6.18-debian-10-r7`
* Added tpl to support external database secrets values
* Added custom volumes/volumesMounts under `common`
* Removed custom volumes from each specific service

### CODE REVIEW

File: stable/xray/templates/xray-statefulset.yaml [hunk 1]
```diff
@@ -189,13 +187,14 @@ spec:
...
-              name: {{ .Values.database.secrets.url.name }}
-              key: {{ .Values.database.secrets.url.key }}
+              name: {{ tpl .Values.database.secrets.url.name . }}
+              key: {{ tpl .Values.database.secrets.url.key . }}
         {{- else if .Values.database.url }}
               name: {{ template "xray.fullname" . }}-database-creds
               key: db-url
         {{- end }}
       {{- end }}
+      {{- if .Values.common.rabbitmq.connectionConfigFromEnvironment }}
```
Comment (danielezer) [inline stable/xray/templates/xray-statefulset.yaml:197]: @chukka why do we need this? Not sure I understand the use case for setting this to false

File: stable/xray/values.yaml [hunk 2]
```diff
@@ -339,8 +345,6 @@ analysis:
   image:
     repository: docker.bintray.io/jfrog/xray-analysis
     # version:
-  updateStrategy: RollingUpdate
```
Comment (danielezer) [inline stable/xray/values.yaml:342]: @chukka  Why did you remove this?

File: stable/xray/templates/xray-statefulset.yaml [hunk 3]
```diff
@@ -1,34 +1,32 @@
...
 kind: StatefulSet
 metadata:
-  name: {{ template "xray-server.fullname" . }}
+  name: {{ template "xray.fullname" . }}
   labels:
     app: {{ template "xray.name" . }}
     chart: {{ template "xray.chart" . }}
     heritage: {{ .Release.Service }}
     release: {{ .Release.Name }}
-    component: {{ .Values.server.name }}
```
Comment (danielezer) [inline stable/xray/templates/xray-statefulset.yaml:10]: @chukka Is there anything that still uses `.Values.server.<SOMETHING>`? I see that it is still in the `values.yaml`

File: stable/xray/values.yaml (hunk 2 above)
Comment (chukka) [reply stable/xray/values.yaml:342]: This is removed and added  here -  https://github.com/jfrog/charts/pull/924/files#diff-82f9672c8009df101fffb76468a8d55fR18 (This is cleanup from xray 2.x charts)

File: stable/xray/templates/xray-statefulset.yaml (hunk 1 above)
Comment (chukka) [reply stable/xray/templates/xray-statefulset.yaml:197]: @danielezer  refer here for more details - https://github.com/jfrog/charts/pull/924/files/2c1886c693cb19b1b2c0fb5c269620ba676eaa21#diff-153bb5de1ec821f30fb4d9703c126d11R284

File: stable/xray/templates/xray-statefulset.yaml (hunk 3 above)
Comment (chukka) [reply stable/xray/templates/xray-statefulset.yaml:10]: Yes  for internal and external ports - https://github.com/jfrog/charts/blob/ecd69d16b3fbdbe58e03e5399f3448e4faca1161/stable/xray/values.yaml#L541

Comment (danielezer) [reply stable/xray/templates/xray-statefulset.yaml:10]: @chukka ok, so please remove everything in the `server` block other than those two
